### PR TITLE
Starting over

### PR DESCRIPTION
**Pull request by Jeff-Duke**
Having an issue where we're trying to connect our action to the container and also to map some fake state,  neither one are connected.  We're connecting /src/containers/Header to /src/component/Header

@stevekinney
@brittanystoroz

### CODE REVIEW

File: src/components/App.js
Comment (stevekinney) [inline src/components/App.js]: First hot take as I dig in a little more. You're requiring `components/Header.js`, but based on your question it sounds like you _want_ `containers/Header.js`—as that's the one you connected to Redux. My suspicion is that if you do the following, then you should be alright.

``` js
import Header from '../containers/Header';
```

Comment (mjvalade) [reply src/components/App.js]: Thanks, that helped. Still not hitting the reducer.

**Comment by Jeff-Duke** [comment]
So we're able to get the stubbed weather and the action into the component, but when we dispatch the action in /components/Header.js via the button click it's not hitting the reducer.

@stevekinney

File: src/index.js
Comment (stevekinney) [inline src/index.js:18]: When I swap it for the following, it works.

``` js
const store = createStore(
  localWeather
);
```

File: src/reducers/rootReducer.js
Comment (stevekinney) [inline src/reducers/rootReducer.js:4]: You don't end up using this anymore—even in `index.js`. It's probably time for this to go.

File: src/index.js
Comment (stevekinney) [inline src/index.js:10]: This is going unused at this point, so it makes sense to remove it before this gets merged.

**Review by stevekinney — CHANGES_REQUESTED** [review]
Before this gets merged, I'd clean up all of the linting errors, most of which are for unused variables.

File: src/index.js
Comment (Jeff-Duke) [reply src/index.js:10]: So the idea of using the rootReducer is that we will likely need additional reducers in the future.  The reason it's not working as written is because of the middleware.  If I remove the applyMiddleware call.  But we're going to need thunk later when we start actually pulling data from the API so what to do?

Comment (stevekinney) [reply src/index.js:10]: Adding a bunch of extra complexity because you _might_ need something down the road is a dangerous path. (In programming, we call this principle "[YAGNI](http://martinfowler.com/bliki/Yagni.html)"). `rootReducer` is not the problem here, it's the middleware stuff. Which is probably just failing because it's not fully implemented. My suggestion is to remove it for now and make it it's own feature and PR in isolation when you're ready for it.

Ideally, we want to make our PRs as small and granular as possible so that they can't hold up other features.

Comment (Jeff-Duke) [reply src/index.js:10]: Agreed, applyMiddleware was put in place by default in the original repo we're all using!

**Pull request closed without merging**
